### PR TITLE
Convert `modules/navigation/newHeaderNavigation` to ES6

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Let's shave the yak first. This converts `modules/navigation/newHeaderNavigation` to ES6, so that we can, in the next step, refactor the code a bit, to finally add the `more` toggle later on. Also it adds flwo type to all methods.

## What is the value of this and can you measure success?

ES6!

## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

tumblr_inline_oa9iltgx0y1raprkq_500

## Tested in CODE?

No.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import fastdom from 'fastdom';
+import ophan from 'ophan/ng';
+import userAccount from 'common/modules/navigation/user-account';
+
+const html = qwery('html')[0];
+const menuItems = qwery('.js-close-nav-list');
+const enhanced = {};
+
+const weShouldEnhance = (checkbox: HTMLInputElement): ?boolean =>
+    !enhanced[checkbox.id] && checkbox && !checkbox.checked;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Would this make more sense:

```
checkbox && !enhanced[checkbox.id] && !checkbox.checked;
```

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    mainListItems.forEach(item => {
+        const listItem = item;
+
+        listItem.style.order = '';
+    });
+};
+
+const toggleSidebar = (event: Event): void => {
+    // #? hacky type cast
+    const button: HTMLElement = (event.target: any);
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Beautiful ❤️

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    const mainMenu = document.getElementById('main-menu');
+    const veggieBurgerLink = qwery('.js-change-link')[0];
+
+    const menuIsOpen = (): boolean =>
+        button.getAttribute('aria-expanded') === 'true';
+
+    if (!mainMenu || !veggieBurgerLink) {
+        return;
```
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Should the return type be `?boolean`

**Review by SiAdcock — APPROVED** [review]
Minor comments, otherwise LGTM 👍

**Review by NataliaLKB — APPROVED** [review]
Awesome!

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
+// @flow
+
+import qwery from 'qwery';
+import fastdom from 'fastdom';
+import ophan from 'ophan/ng';
+import userAccount from 'common/modules/navigation/user-account';
+
+const html = qwery('html')[0];
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: `document.documentElement`?

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    const veggieBurgerLink = qwery('.js-change-link')[0];
+
+    const menuIsOpen = (): boolean =>
+        button.getAttribute('aria-expanded') === 'true';
+
+    if (!mainMenu || !veggieBurgerLink) {
+        return;
+    }
+    if (menuIsOpen()) {
+        fastdom.write(() => {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: How about moving these into function statements, so that they can be optimized? i.e. `fastdom.write(closeMenu)`

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        });
+    }
+};
+
+const applyEnhancementsTo = (checkbox: HTMLElement): void => {
+    fastdom.read(() => {
+        const button = document.createElement('button');
+        const checkboxId = checkbox.id;
+        const checkboxControls = checkbox.getAttribute('aria-controls');
+        const checkboxClasses = Array.prototype.slice.call(checkbox.classList);
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: `const checkboxClasses = ...checkbox.classList` 😃

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js (hunk 4 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Removal of `qwery` will be part of another PR which is already in the pipeline :)

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js (hunk 5 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Also part of another PR.

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js (hunk 6 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: I think `Array.from()` is more readable, but you're right - will be part of the follow-up PR.

Comment (sndrs) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: `const checkboxClasses = [...checkbox.classList]`?

File: static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js (hunk 3 above)
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js]: Good point. I'm going to refactor this in the follow-up PR.
